### PR TITLE
Revise Grammar and update index.md

### PR DESCRIPTION
**Pull request by ComputerLee**
I have revised the first two sections of this article. The previous grammar was poor and needed much improvement to be useful as a educational tool.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #XXXXX

### CODE REVIEW

File: guide/english/computer-science/garbage-collection/index.md [hunk 1]
```diff
@@ -5,16 +5,16 @@ title: Garbage Collection
...
+Garbage collection is the process in which programs try to free up memory space that is no longer used by objects.  Garbage collection is implemented differently for every language.  Most high-level programming languages have some sort of garbage collection built it.  Low-level programming languages may add garbage collection through libraries.
 
-As said above, every programming language has their own way of GC. In C programming, developers need to take care of memory allocation and deallocation using malloc() and dealloc() functions. But, in case of C# developers no need to take care of GC and it's not recommended as well.    
+As said above, every programming language has their own way of implementing GC. In C programming, developers need to take care of memory allocation and deallocation using malloc() and dealloc() functions. But, in case of C# developers do not need to take address GC and it's also not a recommended practice.
```
Comment (co16353sidak) [inline guide/english/computer-science/garbage-collection/index.md]: on this line, it should be either `need to address GC` or it should be reverted back to the previous like `need to take care of GC`.
`need to take address GC` just seems odd and I feel it is grammatically wrong.

File: guide/english/computer-science/garbage-collection/index.md [hunk 2]
```diff
@@ -5,16 +5,16 @@ title: Garbage Collection
...
  
  #### How memory allocation happening?
- In C#, memory allocation of objects are happened in managed heap. which takes care by CLR (common language runtime). Memory allocation for the heap is done through win32 dll in OS as like in the C. But, In C objects are placed in memory where ever the free space suits the size of object. And, the memory mapping is woks based on Linkedlist concepts. In C#, memory allocation for heap is happening in linear manner. like one after another. 
+ In C#, memory allocation of objects are addressed in managed heap. which takes care by CLR (common language runtime). Memory allocation for the heap is done through win32 dll in the Operating System just like in C. But, In C objects are placed in memory where ever the free space suits the size of object. And, the memory mapping is works based on Linkedlist concepts. In C#, memory allocation for heap is happening in a consecutive manner.
```
Comment (co16353sidak) [inline guide/english/computer-science/garbage-collection/index.md]: removing the word `works` will be better here: `memory mapping is works based` -> `memory mapping is based`
small grammatical change: `Is happening in a consecutive manner` -> `for heap occurs in a consecutive manner`.

**Review by theNaut1lus — CHANGES_REQUESTED** [review]
@ComputerLee small grammatical suggestions I have mentioned down in the comments, please consider these to improve the readability of your changes.

/cc: @RandellDawson merge conflict present, can be merged post resolution and changes.

**Comment by co16353sidak** [comment]
@ComputerLee consider removing merge conflicts of your PR with the `master` branch

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@ComputerLee See the changes requested by @co16353sidak.  If you need help resolving the merge conflict, just ask.

Thank you.

**Comment by RandellDawson** [comment]
Hi @ComputerLee 

As there has been no response from your end for a while, we are closing this PR.

Feel free to request this PR to be reopened or create a new PR, if you still wish to make the changes you originally added plus any additional change requests asked of you.  Also, remember to update your fork, so you have the latest version of master in case it has changed significantly.

Happy Coding!

**Pull request closed without merging**